pagopa/io-app
# fix: [IA-769] Rptid custom field not filled or filled with wrong value

## THIS PR DEPENDS ON https://github.com/pagopa/io-react-native-zendesk/pull/22
## Short description
This PR fixes the bug that leads to a wrong `Rptid` value in the `BlockedPaymentRptIdId` Zendesk custom field.

## List of changes proposed in this pull request
- Added the `BlockedPaymentRptIdId` filling in the `PaymentHistoryDetailsScreen`
- Updated the `io-react-native-zendesk` to the `3.0.21` version
- Modified the `resetAssistanceData` using the new specific reset functions
- Reset the custom field when the user starts the assistance flow from the `?` button

## How to test
Try the following scenarios:
- Open an assistance request from a failed payment (`TransactionErrorScreen`) -> The ticket should show the `Rptid` in the `BlockedPaymentRptIdId` Zendesk custom field.
   - Open an assistance request from the `?` button -> The ticket should show the `BlockedPaymentRptIdId` Zendesk custom field void
- Close and open the app then open an assistance request from the payment history in the FAQ (`PaymentHistoryDetailsScreen`) -> The ticket should show the `Rptid` in the `BlockedPaymentRptIdId` Zendesk custom field.
   - Open an assistance request from the `?` button -> The ticket should show the `BlockedPaymentRptIdId` Zendesk custom field void
